src/extension.c

```diff
@@ -56,20 +56,33 @@ enum ExtensionState
 };
 
 static enum ExtensionState extstate = EXTENSION_STATE_INITIAL;
-static TransactionId drop_transaction_id = InvalidTransactionId;
 
 bool
 extension_is_being_dropped(Oid relid)
 {
+	/*
+	 * during a extension drop, the extension proxy table will be dropped.
+	 * This will automatically cause a relcache invalidation on the proxy
+	 * tables's OID.
+	 */
 	return relid == extension_proxy_oid;
 }
 
+
+static bool
+proxy_table_exists()
```

Comment:
So from what I could tell, the postgres code is pretty good at making sure dependent objects are dropped before the objects they depend on. So, I think relying on this is pretty safe. So the flow, as I see it is: the proxy table is dropped (which causes a relcache invalidaton of the proxy table internally, more on that below). The relcache invalidation goes to all backends which will reset the extension. Now, as long as the extension cannot go back to CREATED while the drop is in process, we are safe. That's what the `proxy_table_exists()` check guards against.